test/e2e/predictor/test_response_headers.py

```diff
@@ -0,0 +1,193 @@
+        containers=[
+            V1Container(
+                name="kserve-container",
+                image="kserve/custom-model-grpc:" + os.environ.get("GITHUB_SHA"),
+                # Override the entrypoint to run the custom model rest server
+                command=["python", "-m", "custom_model.model"],
+                resources=V1ResourceRequirements(
+                    requests={"cpu": "50m", "memory": "128Mi"},
+                    limits={"cpu": "100m", "memory": "1Gi"},
+                ),
+                ports=[V1ContainerPort(container_port=8080, protocol="TCP")],
+                args=["--model_name", model_name],
+            )
+        ]
+    )
+
+    transformer = V1beta1TransformerSpec(
+        containers=[
+            V1Container(
+                name="kserve-container",
+                image="kserve/custom-image-transformer-grpc:"
+                + os.environ.get("GITHUB_SHA"),
+                # Override the entrypoint to run the custom transformer rest server
+                command=["python", "-m", "custom_transformer.model"],
+                resources=V1ResourceRequirements(
+                    requests={"cpu": "50m", "memory": "128Mi"},
+                    limits={"cpu": "100m", "memory": "1Gi"},
+                ),
+                args=["--model_name", model_name, "--predictor_protocol", "v1"],
+            )
+        ]
+    )
+
+    isvc = V1beta1InferenceService(
+        api_version=constants.KSERVE_V1BETA1,
+        kind=constants.KSERVE_KIND,
+        metadata=client.V1ObjectMeta(
+            name=service_name, namespace=KSERVE_TEST_NAMESPACE
+        ),
+        spec=V1beta1InferenceServiceSpec(predictor=predictor, transformer=transformer),
+    )
+
+    kserve_client = KServeClient(
+        config_file=os.environ.get("KUBECONFIG", "~/.kube/config")
+    )
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    isvc = kserve_client.get(
+        service_name,
+        namespace=KSERVE_TEST_NAMESPACE,
+        version=constants.KSERVE_V1BETA1_VERSION,
+    )
+    cluster_ip, host, path = get_isvc_endpoint(isvc)
+    headers = {"Host": host, "Content-Type": "application/json"}
+
+    if model_name is None:
+        model_name = service_name
+
+    url = f"http://{cluster_ip}{path}/v1/models/{model_name}:predict"
+
+    time.sleep(10)
+    with open(input_json) as json_file:
+        data = json.load(json_file)
+        response = requests.post(url, json.dumps(data), headers=headers)
+        logging.info(
+            "Got response code %s, content %s", response.status_code, response.content
+        )
+        if response.status_code == 200:
+            res_data = json.loads(response.content.decode("utf-8"))
+            res_data["headers"] = response.headers
+        else:
+            response.raise_for_status()
+
+    response_headers = res_data["headers"]
+    assert response_headers["my-header"] == "test_header"
```

Comment:
What kind of information should we expect in the response headers?
Not sure if it is a good test, since you forced this header to be added.